From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ssch71 <ssch71@users.noreply.github.com>
Date: Tue, 20 Oct 2026 07:40:00 +0900
Subject: [PATCH] Add Galaxy Z Flip5 camera resolutions

The S23 camera app has no entries for the Flip5 sensor crop and 22:9
sizes in camera-feature.xml, so getResolution() throws on launch.
Add them, mapping 22:9 to the closest existing full ratio.
---
diff --git a/smali_classes2/l9/a2.smali b/smali_classes2/l9/a2.smali
index 19a327c..95a7930 100644
--- a/smali_classes2/l9/a2.smali
+++ b/smali_classes2/l9/a2.smali
@@ -516,6 +516,28 @@
 
 .field public static final enum RESOLUTION_960X960:Ll9/a2;
 
+.field public static final enum RESOLUTION_1264X2232:Ll9/a2;
+
+.field public static final enum RESOLUTION_1280X2268:Ll9/a2;
+
+.field public static final enum RESOLUTION_1440X1504:Ll9/a2;
+
+.field public static final enum RESOLUTION_2080X2080:Ll9/a2;
+
+.field public static final enum RESOLUTION_2640X1080:Ll9/a2;
+
+.field public static final enum RESOLUTION_2784X2080:Ll9/a2;
+
+.field public static final enum RESOLUTION_2880X2992:Ll9/a2;
+
+.field public static final enum RESOLUTION_3008X1232:Ll9/a2;
+
+.field public static final enum RESOLUTION_3008X1696:Ll9/a2;
+
+.field public static final enum RESOLUTION_3968X1624:Ll9/a2;
+
+.field public static final enum RESOLUTION_4000X1638:Ll9/a2;
+
 .field private static final mIdMap:Ljava/util/Map;
     .annotation system Ldalvik/annotation/Signature;
         value = {
@@ -4774,6 +4796,226 @@
 
     sput-object v33, Ll9/a2;->RESOLUTION_4640X1988:Ll9/a2;
 
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_1264X2232"
+
+    const/16 v2, 0xff
+
+    const/16 v3, 0x11c
+
+    const/16 v4, 0x4f0
+
+    const/16 v5, 0x8b8
+
+    sget-object v6, Ll9/b;->RATIO_16x9:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_1264X2232:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_1280X2268"
+
+    const/16 v2, 0x100
+
+    const/16 v3, 0x11d
+
+    const/16 v4, 0x500
+
+    const/16 v5, 0x8dc
+
+    sget-object v6, Ll9/b;->RATIO_16x9:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_1280X2268:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_1440X1504"
+
+    const/16 v2, 0x101
+
+    const/16 v3, 0x11e
+
+    const/16 v4, 0x5a0
+
+    const/16 v5, 0x5e0
+
+    sget-object v6, Ll9/b;->RATIO_1x1:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_1440X1504:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_2080X2080"
+
+    const/16 v2, 0x102
+
+    const/16 v3, 0x11f
+
+    const/16 v4, 0x820
+
+    const/16 v5, 0x820
+
+    sget-object v6, Ll9/b;->RATIO_1x1:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_2080X2080:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_2640X1080"
+
+    const/16 v2, 0x103
+
+    const/16 v3, 0x120
+
+    const/16 v4, 0xa50
+
+    const/16 v5, 0x438
+
+    sget-object v6, Ll9/b;->RATIO_21DOT2x9:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_2640X1080:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_2784X2080"
+
+    const/16 v2, 0x104
+
+    const/16 v3, 0x121
+
+    const/16 v4, 0xae0
+
+    const/16 v5, 0x820
+
+    sget-object v6, Ll9/b;->RATIO_4x3:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_2784X2080:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_2880X2992"
+
+    const/16 v2, 0x105
+
+    const/16 v3, 0x122
+
+    const/16 v4, 0xb40
+
+    const/16 v5, 0xbb0
+
+    sget-object v6, Ll9/b;->RATIO_1x1:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_2880X2992:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_3008X1232"
+
+    const/16 v2, 0x106
+
+    const/16 v3, 0x123
+
+    const/16 v4, 0xbc0
+
+    const/16 v5, 0x4d0
+
+    sget-object v6, Ll9/b;->RATIO_21DOT2x9:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_3008X1232:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_3008X1696"
+
+    const/16 v2, 0x107
+
+    const/16 v3, 0x124
+
+    const/16 v4, 0xbc0
+
+    const/16 v5, 0x6a0
+
+    sget-object v6, Ll9/b;->RATIO_16x9:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_3008X1696:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_3968X1624"
+
+    const/16 v2, 0x108
+
+    const/16 v3, 0x125
+
+    const/16 v4, 0xf80
+
+    const/16 v5, 0x658
+
+    sget-object v6, Ll9/b;->RATIO_21DOT2x9:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_3968X1624:Ll9/a2;
+
+    new-instance v0, Ll9/a2;
+
+    const-string v1, "RESOLUTION_4000X1638"
+
+    const/16 v2, 0x109
+
+    const/16 v3, 0x126
+
+    const/16 v4, 0xfa0
+
+    const/16 v5, 0x666
+
+    sget-object v6, Ll9/b;->RATIO_21DOT2x9:Ll9/b;
+
+    const/16 v7, 0x1e
+
+    invoke-direct/range {v0 .. v7}, Ll9/a2;-><init>(Ljava/lang/String;IIIILl9/b;I)V
+
+    sput-object v0, Ll9/a2;->RESOLUTION_4000X1638:Ll9/a2;
+
     invoke-static {}, Ll9/a2;->b()[Ll9/a2;
 
     move-result-object v0
@@ -4914,7 +5156,7 @@
 .method public static synthetic b()[Ll9/a2;
     .locals 3
 
-    const/16 v0, 0xff
+    const/16 v0, 0x10a
 
     new-array v0, v0, [Ll9/a2;
 
@@ -6448,6 +6690,72 @@
 
     aput-object v1, v0, v2
 
+    sget-object v1, Ll9/a2;->RESOLUTION_1264X2232:Ll9/a2;
+
+    const/16 v2, 0xff
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_1280X2268:Ll9/a2;
+
+    const/16 v2, 0x100
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_1440X1504:Ll9/a2;
+
+    const/16 v2, 0x101
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_2080X2080:Ll9/a2;
+
+    const/16 v2, 0x102
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_2640X1080:Ll9/a2;
+
+    const/16 v2, 0x103
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_2784X2080:Ll9/a2;
+
+    const/16 v2, 0x104
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_2880X2992:Ll9/a2;
+
+    const/16 v2, 0x105
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_3008X1232:Ll9/a2;
+
+    const/16 v2, 0x106
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_3008X1696:Ll9/a2;
+
+    const/16 v2, 0x107
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_3968X1624:Ll9/a2;
+
+    const/16 v2, 0x108
+
+    aput-object v1, v0, v2
+
+    sget-object v1, Ll9/a2;->RESOLUTION_4000X1638:Ll9/a2;
+
+    const/16 v2, 0x109
+
+    aput-object v1, v0, v2
+
     return-object v0
 .end method
 
-- 
2.47.0
